src/IStakeManager.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ITrustedCodehashAccess } from "./access/ITrustedCodehashAccess.sol";
+
+interface IStakeManager is ITrustedCodehashAccess {
+    error StakeManager__FundsLocked();
+    error StakeManager__InvalidLockTime();
+    error StakeManager__InsufficientFunds();
+    error StakeManager__StakeIsTooLow();
+
+    function stake(uint256 _amount, uint256 _seconds) external;
+    function unstake(uint256 _amount) external;
+    function lock(uint256 _secondsIncrease) external;
+    function leave() external returns (bool _leaveAccepted);
+    function acceptUpdate() external returns (address _migrated);
```

Comment:
I would like to leave it here in PR, we don't have to implement it yet, but just define the interface.

It has a return (address _migrated) which in case of upgradabilty through proxy, it would return the address of itself. In case the upgradabiltiy is to a new address, it would return the new address - so this manage the upgradability for migration style and proxy style.